js/lex.go

```diff
@@ -33,6 +33,30 @@ const (
 	TemplateToken
 )
 
+// TokenState determines a state in which next token should be read
+type TokenState uint32
```

Comment:
Ah yeah makes sense. But then `TokenType` across the package uses `uint32` everywhere, so tried to be consistent with types for enum.